tests/api/mam_test.c

```diff
@@ -1,62 +1,213 @@
+  char seed[NUM_TRYTES_ADDRESS + 1] = {};
+  const char* json_template_send = "{\"x-api-key\":\"" TEST_TOKEN "\",\"data\":{\"seed\":\"%s\",\"ch_mss_depth\":" STR(
+      TEST_CH_DEPTH) ",\"message\":\"%s:%d\"}, \"protocol\":\"MAM_V1\"}";
+  const char payload[] = "This is test payload number";
+  const int len = strlen(json_template_send) + NUM_TRYTES_ADDRESS + strlen(payload) + 2;
+  gen_rand_seed(seed);
+  for (int i = 0; i < msg_num; i++) {
+    char* json_result;
+    mam_res_array[i] = send_mam_res_new();
+    char* json = (char*)malloc(sizeof(char) * len);
+    snprintf(json, len, json_template_send, seed, payload, i);
+    TEST_ASSERT_EQUAL_INT32(
+        SC_OK, api_send_mam_message(&ta_core.ta_conf, &ta_core.iota_conf, &ta_core.iota_service, json, &json_result));
+    send_mam_res_deserialize(json_result, mam_res_array[i]);
+    free(json);
+    free(json_result);
+  }
+
+  // The current chid1 should be equal to the chid of next channel. Element with index `channel_leaf_msg_num` is the
+  // last element of the current channel.
+  TEST_ASSERT_EQUAL_STRING(mam_res_array[channel_leaf_msg_num]->chid, mam_res_array[channel_leaf_msg_num - 1]->chid1);
+
+  const char* json_template_recv = "{\"data_id\":{\"chid\":\"%s\"},\"protocol\":\"MAM_V1\"}";
+  const int json_size = sizeof(char) * (strlen(json_template_recv) + NUM_TRYTES_ADDRESS);
+  for (int i = 0; i < complete_ch_num + 1; i++) {
+    char *json = NULL, *json_result = NULL;
+    json = (char*)malloc(json_size);
+    snprintf(json, json_size, json_template_recv, mam_res_array[i * channel_leaf_msg_num]->chid);
+    TEST_ASSERT_EQUAL_INT32(SC_OK, api_recv_mam_message(&ta_core.iota_conf, &ta_core.iota_service, json, &json_result));
+    for (int j = i * channel_leaf_msg_num; j < ((i + 1) * channel_leaf_msg_num) && (j < msg_num); j++) {
+      // Check whether a message exist under assigning channel ID.
+      char substr[strlen(payload) + 3];
+      snprintf(substr, strlen(payload) + 3, "%s:%d", payload, j);
+      TEST_ASSERT_TRUE(strstr(json_result, substr));
+    }
+
+    // Deserialize the response, and compare whether the payload list contains correct number of message under a certain
+    // channel ID.
+    int len = 0;
+    cJSON *json_obj = cJSON_Parse(json_result), *current_obj = NULL;
+    cJSON* json_item = cJSON_GetObjectItemCaseSensitive(json_obj, "payload");
+    TEST_ASSERT_TRUE(cJSON_IsArray(json_item));
+    cJSON_ArrayForEach(current_obj, json_item) {
+      if (current_obj->valuestring != NULL) {
+        len++;
+      }
+    }
+    if (i == complete_ch_num) {
+      TEST_ASSERT_EQUAL_INT16(dangling_msg_num, len);
+    } else {
+      TEST_ASSERT_EQUAL_INT16(channel_leaf_msg_num, len);
+    }
+    cJSON_Delete(json_obj);
+    free(json);
+    free(json_result);
+  }
+
+  for (int i = 0; i < msg_num; i++) {
+    send_mam_res_free(&(mam_res_array[i]));
+  }
+}
+
+int main(int argc, char* argv[]) {
   UNITY_BEGIN();
+  init_rand_seed();
```

Comment:
I am confused about the name scheme. Should we use `init_rand_seed` or `rand_seed_init`?